pkg/apimachinery/identity/requester.go

```diff
@@ -15,13 +15,15 @@ type Requester interface {
 
 	// GetIdentityType returns the type for the requester
 	GetIdentityType() claims.IdentityType
+	// IsIdentityType returns true if identity type for requester matches any expected identity type
+	IsIdentityType(expected ...claims.IdentityType) bool
 	// GetRawIdentifier returns only the identifier part of the UID, excluding the type
 	GetRawIdentifier() string
-	// Deprecated: use GetUID instead
+	// GetInternalID returns only the identifier part of the ID, excluding the type
 	GetInternalID() (int64, error)
 	// GetID returns namespaced internalID for the entity
 	// Deprecated: use GetUID instead
-	GetID() TypedID
+	GetID() string
```

Comment:
do we need this one?    can use use GetInternalID or GetIdentityType instead?

Reply:
I think we can, but I would rather do that in a follow up pr, there are many call sites to update to get rid of it.